pydatastructs/trees/suffix_tree.py

```diff
@@ -0,0 +1,260 @@
+__all__ = [
+    'SuffixTree'
+]
+
+class Suffix_Node():
```

Comment:
This class should be in https://github.com/codezonediitj/pydatastructs/blob/master/pydatastructs/utils/misc_util.py and should inherit `Node` class.